### PR TITLE
Add fake client that cleanly implements the Client interface instead of using GRPC libraries.

### PR DESCRIPTION
**Pull request by JoshuaKrstic**
Add fake client that cleanly implements the Client interface instead of using GRPC libraries.

### CODE REVIEW

**Review by josephlr — COMMENTED** [review]
Can we not just delete `fake_attestationverifier.go` and `fake_attestationverifier_test.go` in this PR?

File: launcher/verifier/fake/fakeverifier.go [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+	// Determine signing algorithm.
+	signingMethod := jwt.SigningMethodRS256
+	now := jwt.TimeFunc()
+	claims := jwt.RegisteredClaims{
+		IssuedAt:  &jwt.NumericDate{Time: now},
+		NotBefore: &jwt.NumericDate{Time: now},
+		ExpiresAt: &jwt.NumericDate{Time: now.Add(60 * 60 * 1e9)}, // Add takes nanoseconds
+		Audience:  []string{"TestingAudience"},
+		Issuer:    "TestingIssuer",
+		Subject:   "TestingSubject",
```
Comment (josephlr) [inline launcher/verifier/fake/fakeverifier.go:47]: Can we have these fields also contain real-ish data?

- `Audiance:` `"https://sts.googleapis.com/"`
- `Issuer:` `"https://confidentialcomputing.googleapis.com/"`
- `Subject:` `"https://www.googleapis.com/compute/v1/projects/fakeProject/zones/fakeZone/instances/fakeInstance"`

File: launcher/verifier/fake/fakeverifier.go [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+}
+
+// CreateChallenge returns a hard coded, basic challenge.
+//
+// If you have found this method is insufficient for your tests, this class must be updated to
+// allow for better testing.
+func (fc *fakeClient) CreateChallenge(ctx context.Context) (*verifier.Challenge, error) {
+	return &verifier.Challenge{
+		Name:  "FakeName",
+		Nonce: []byte{0x0},
```
Comment (josephlr) [inline launcher/verifier/fake/fakeverifier.go:29]: Can we have this Name and nonce at least look something like real data?

For example:
  - `Name:` `"projects/fakeProject/locations/fakeRegion/challenges/d882c62f-452f-4709-9335-0cccaf64eee1"`
  - `Nonce:` 16 bytes of random data

Comment (JoshuaKrstic) [reply launcher/verifier/fake/fakeverifier.go:29]: Sure, let me know what you think of this

File: launcher/verifier/fake/fakeverifier.go [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+// If you have found this method is insufficient for your tests, this class must be updated to
+// allow for better testing.
+func (fc *fakeClient) VerifyAttestation(ctx context.Context, request verifier.VerifyAttestationRequest) (*verifier.VerifyAttestationResponse, error) {
+	// Determine signing algorithm.
+	signingMethod := jwt.SigningMethodRS256
+	now := jwt.TimeFunc()
+	claims := jwt.RegisteredClaims{
+		IssuedAt:  &jwt.NumericDate{Time: now},
+		NotBefore: &jwt.NumericDate{Time: now},
+		ExpiresAt: &jwt.NumericDate{Time: now.Add(60 * 60 * 1e9)}, // Add takes nanoseconds
```
[the remark this answers is not in the document]
Comment (JoshuaKrstic) [reply launcher/verifier/fake/fakeverifier.go:44]: Great

**Review by josephlr — APPROVED** [review]
Nice job on this

**Pull request merged**